### PR TITLE
UI: upgrade HDS to `4.12.0`

### PR DESCRIPTION
**Pull request by hellobontempo**
### Description
Upgrades hds to latest version, this also handles some deprecations coming in `5.0.0`

1. The `Hds::SideNav::Header::IconButton` sub-component is now deprecated and will be removed in the next major version of `@hashicorp/design-system-components`. Use `Hds::Button` with the `isIconOnly` variant instead.
2.  **This change broke a lot of tests, which is why there are so many changes in this PR**: The `@text` argument for \"Hds::Dropdown::ListItem::Interactive\" has been deprecated. Please put text in the yielded block. [deprecation id: hds.dropdown.list-item.interactive] This will be removed in @hashicorp/design-system-components 5.0.0. See https://helios.hashicorp.design/components/dropdown?tab=version%20history#4100 for more details
        

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this PR is in the ENT repo and needs to be backported, backport  
  to N, N-1, and N-2, using the `backport/ent/x.x.x+ent` labels. If this PR is in the CE repo, you should only backport to N, using the `backport/x.x.x` label, not the enterprise labels.
    - [ ] If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

**Comment by hellobontempo** [comment]
✅  enterprise passed
Screenshot 2024-09-26 at 2 55 57 PM

File: ui/app/controllers/vault/cluster/access/oidc/clients/client.js [hunk 1]
```diff
@@ -5,22 +5,7 @@
 
 import Controller from '@ember/controller';
 import { service } from '@ember/service';
-import { tracked } from '@glimmer/tracking';
 
 export default class OidcClientController extends Controller {
   @service router;
-  @tracked isEditRoute;
```
Comment (hellobontempo) [inline ui/app/controllers/vault/cluster/access/oidc/clients/client.js:12]: This was causing a strange error in login tests? The constructor was being called in an unrelated, oidc auth login test. It seemed like the controller was being used when it shouldn't have been. So I refactored to a better pattern of just checking the route in the template which is how I refactored this.

File: ui/app/styles/core/element-styling.scss [hunk 2]
```diff
@@ -195,8 +195,3 @@ label {
     cursor: not-allowed;
   }
 }
-
-// TODO remove when HDS has released fix
-.hds-form-masked-input .hds-form-masked-input__toggle-button {
```
Comment (hellobontempo) [inline ui/app/styles/core/element-styling.scss:200]: 🤣 this was the whole reason for this upgrade....there was some scope creep a bit. Although these deprecations I resolved weren't going to be a problem until 5.0 I thought I'd get ahead of them now!

File: ui/app/templates/components/generated-item-list.hbs [hunk 3]
```diff
@@ -79,20 +79,17 @@
             data-test-popup-menu-trigger
           />
           <dd.Interactive
-            @text="View {{singularize @itemType}}"
```
Comment (hellobontempo) [inline ui/app/templates/components/generated-item-list.hbs:82]: Extra eyes making sure there's no copy pasta here would be helpful! All of the `@text` args should now be in the yielded block for any `Interactive` elements

File: ui/app/templates/vault/cluster/access/oidc/clients/client.hbs [hunk 4]
```diff
@@ -3,7 +3,7 @@
   SPDX-License-Identifier: BUSL-1.1
 ~}}
 
-{{#if this.showHeader}}
```
Comment (hellobontempo) [inline ui/app/templates/vault/cluster/access/oidc/clients/client.hbs:6]: this is the refactor removing the showHeader getter from the OidcClientController3

File: ui/app/templates/vault/cluster/access/oidc/keys/index.hbs [hunk 5]
```diff
@@ -21,7 +21,7 @@
       <div class="level-left">
         <div>
           <Icon @name="key" class="has-text-grey-light" />
-          <span class="has-text-weight-semibold is-underline">
+          <span class="has-text-weight-semibold is-underline" data-test-item>
```
Comment (hellobontempo) [inline ui/app/templates/vault/cluster/access/oidc/keys/index.hbs:24]: selectors have to be more specific because `innerText` on linked block dom elements now include the `Interactive` text

File: ui/app/styles/components/popup-menu.scss [hunk 6]
```diff
@@ -55,7 +55,8 @@
     &:hover {
       background-color: $ui-gray-050;
     }
-    div {
```
Comment (hellobontempo) [inline ui/app/styles/components/popup-menu.scss:58]: the misalignment had been bothering me, so decided to fix

# before/after
Screenshot 2024-09-27 at 12 05 32 PM

File: ui/tests/acceptance/secrets/backend/kv/secret-test.js [hunk 7]
```diff
@@ -20,6 +20,7 @@ import { runCmd } from 'vault/tests/helpers/commands';
 import { PAGE } from 'vault/tests/helpers/kv/kv-selectors';
 import codemirror from 'vault/tests/helpers/codemirror';
 import { GENERAL } from 'vault/tests/helpers/general-selectors';
+import { SECRET_ENGINE_SELECTORS as SS } from 'vault/tests/helpers/secret-engine/secret-engine-selectors';
```
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/kv/secret-test.js:23]: NIT: I've been calling these SES instead of SS.  
image

**Review by Monkeychip — APPROVED** [review]
Nothing blocking. Nice work on this and thank you for tackling early in the release cycle.

File: ui/app/components/sidebar/frame.hbs [hunk 8]
```diff
@@ -18,9 +18,11 @@
             />
           </:logo>
           <:actions>
-            <Hds::SideNav::Header::IconButton
+            <Hds::Button
+              @isIconOnly={{true}}
+              @size="large"
               @icon="terminal-screen"
-              @ariaLabel="Console toggle"
+              @text="Console toggle"
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/components/sidebar/frame.hbs:25]: I think it makes sense that the button has to have `@text` and that it's used for the aria, but maybe because of all the button work I've done 😂

Thanks for quoting the docs!
